.github/workflows/python2-3-app.yml

```diff
@@ -0,0 +1,59 @@
+name: DB12 testing
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.6, 3.9, 2.7]
+        
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+  
+    - uses: actions/checkout@v2
+    - uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        activate-environment: python
+        environment-file: environment.yml
+        
+    - name: Pylint
+      run: |
+        source /usr/share/miniconda/etc/profile.d/conda.sh
+        conda activate python
+        pylint tests/test.py --max-statements=100
+        pylint src/DIRACbenchmark/dirac_benchmark.py
+        
+    - name: Test with pytest
+      run: |
+        source /usr/share/miniconda/etc/profile.d/conda.sh
+        conda activate python
+        pytest tests/test.py
+        
+    - name: Install pypa/build
+      run: >-
+        python -m
+        pip install
+        build
+        --user
+        
+    - name: Build a binary wheel and a source tarball
+      run: >-
+        python -m
+        build
+        --sdist
+        --wheel
+        --outdir dist/
+        .
+    - uses: fkirc/skip-duplicate-actions@master
+    - name: Publish distribution to PyPI
+      if: startsWith(github.ref, 'refs/tags')
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        skip_existing: 'true'
+        password: ${{ secrets.PYPI_API_TOKEN }}
```

Comment:
Seems like this part (line 38 to 59) is executed 3 times because there are 3 different python versions used here.
I think it can be done only once: maybe adding this part in another github actions workflow could solve the problem.